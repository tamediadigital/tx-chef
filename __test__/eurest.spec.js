const fn = require('../helpers/eurest');

const DEBUG = true;

describe('get eurest menu data', () => {
  test('exports a promise with menu data', () => {
    expect.assertions(2);
    return fn('url', DEBUG).then(data => {
      expect(data.date).toBe('20.09.22');
      expect(data).toMatchSnapshot();
    });
  });
  
  test('gets first meal', () => {
    expect.assertions(4);
    return fn('url', DEBUG).then(data => {
      const meal = Object.values(data.meals).filter(mealObj => mealObj.category === 'Brasserie').pop();
  
      expect(meal.title).toBe('Ghackets (CH)');
      expect(meal.description).toBe('und Hörnli mit Reibkäse und Apfelmus');
      expect(meal.prices[0]).toBe('CHF 9,50');
      expect(meal.prices[1]).toBe('Ext. 13,50');
    });
  });
});
