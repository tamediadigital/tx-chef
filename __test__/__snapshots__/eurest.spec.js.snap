// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get eurest menu data exports a promise with menu data 1`] = `
Object {
  "date": "20.09.22",
  "meals": Object {
    "0": Object {
      "category": "Brasserie",
      "description": "und Hörnli mit Reibkäse und Apfelmus",
      "id": "0",
      "prices": Array [
        "CHF 9,50",
        "Ext. 13,50",
        "669 kcal",
      ],
      "title": "Ghackets (CH)",
    },
    "1": Object {
      "category": "Grün und natürlich",
      "description": "mit Pitabrot und Joghurt",
      "id": "1",
      "prices": Array [
        "CHF 9,50",
        "Ext. 13,50",
        "539 kcal",
      ],
      "title": "Marokkanisches Safran Gemüse",
    },
    "2": Object {
      "category": "Tagessuppe",
      "description": null,
      "id": "2",
      "prices": Array [
        "CHF 1,40",
        "Ext. 1,70",
      ],
      "title": "Karotten-Ingwersuppe",
    },
  },
}
`;
